fe/fe-core/src/main/java/com/starrocks/alter/Alter.java

```diff
@@ -473,6 +495,34 @@ private void processRenameMaterializedView(String oldMvName, String newMvName, D
         LOG.info("rename materialized view[{}] to {}, id: {}", oldMvName, newMvName, materializedView.getId());
     }
 
+    private void processChangeMaterializedViewStatus(MaterializedView materializedView, String status) {
+        if ("active".equalsIgnoreCase(status)) {
+            String viewDefineSql = materializedView.getViewDefineSql();
+            ConnectContext context = new ConnectContext();
+            context.setQualifiedUser(AuthenticationManager.ROOT_USER);
+            context.setCurrentUserIdentity(UserIdentity.ROOT);
+            context.setCurrentRoleIds(Sets.newHashSet(PrivilegeBuiltinConstants.ROOT_ROLE_ID));
+
+            List<StatementBase> statementBaseList = SqlParser.parse(viewDefineSql, context.getSessionVariable());
+            QueryStatement queryStatement = (QueryStatement) statementBaseList.get(0);
+            try {
+                Analyzer.analyze(queryStatement, context);
+            } catch (SemanticException e) {
+                throw new SemanticException("Can not active materialized view [" + materializedView.getName() +
+                        "] because analyze materialized view define sql: \n\n" + viewDefineSql +
+                        "\n\nCause an error: " + e.getDetailMsg());
+            }
+
+            Map<TableName, Table> tableNameTableMap = AnalyzerUtils.collectAllConnectorTableAndView(queryStatement);
+            List<BaseTableInfo> baseTableInfos = MaterializedViewAnalyzer.getBaseTableInfos(tableNameTableMap);
+            materializedView.setBaseTableInfos(baseTableInfos);
+            materializedView.getRefreshScheme().getAsyncRefreshContext().clearVisibleVersionMap();
```

Comment:
The historical data of the MV can be queried, and theoretically it can be forced to trigger another refresh. This issue can be discussed together with the next PR issue.

Reply:
After set active, we need to refresh this mv to enable the query rewrite?